src/sympc/protocol/falcon/falcon.py

```diff
@@ -404,3 +405,52 @@ def multiplication_protocol(
             x.ring_size,
         )
         return z_value
+
+    @staticmethod
+    def select_shares(x: MPCTensor, y: MPCTensor, b: MPCTensor) -> MPCTensor:
+        """Returns either x or y based on bit b.
+
+        Args:
+            x (MPCTensor): input tensor
+            y (MPCTensor): input tensor
+            b (MPCTensor): input tensor which is shares of a bit used as selector bit.
+
+        Returns:
+            z (MPCTensor):Returns x (if b==0) or y (if b==1).
+
+        Raises:
+            ValueError: If the selector bit tensor is not of ring size "2".
+        """
+        ring_size = int(b.share_ptrs[0].get_ring_size().get_copy())
+        shape = b.shape
+        if ring_size != 2:
+            raise ValueError(
+                f"Invalid {ring_size} for bit_injection,must be of ring size 2"
+            )
+        if shape is None:
+            raise ValueError("The selector bit tensor must have a valid shape.")
+        session = x.session
+
+        # TODO: Should be made to generate with CryptoProvider in Preprocessing stage.
+        c_ptrs: List[ReplicatedSharedTensor] = []
+        for session_ptr in session.session_ptrs:
+            c_ptrs.append(
+                session_ptr.prrs_generate_random_share(
+                    shape=shape, ring_size=str(ring_size)
+                )
+            )
+
+        c = MPCTensor(shares=c_ptrs, session=session, shape=shape)  # bit random share
+        c_r = ABY3.bit_injection(
+            c, session, session.ring_size
+        )  # bit random share in session ring.
+
+        tensor_type = get_type_from_ring(session.ring_size)
+        mask = (b + c).reconstruct(decode=False).type(tensor_type)
+
```

Comment:
Also, I think it would be good to still do
if mask=1:
   d = (1-c)
else:
   d = c

It's computationally lesser expensive since we are using public operation. Doing it mathematically involves private operations making it expensive. Select shares is used in several operations such it would be good to get as much compute efficiency as possible.

Reply:
Good question, Hrishi, we have to do it different for our tensor setting, the paper method won't work, I initially implemented what you suggested, what we want really want is ,given `mpctensor`  `x` and `y` , we want a mix of values(some from x and some from y) ,depending on the selector bit tensor.